enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,385 @@
+specific upstream branch that was mirrored. Only tagged commits will ever be
+brought into the release branches.
+
+#### Syncing
+
+Today, we sync `operator-framework/operator-sdk` into
+`openshift/ocp-release-operator-sdk`. We sync it manually with `git merge`. The
+`master` branch downstream currently tracks the latest OpenShift release. So
+when moving to a new major SDK release downstream, the `master` branch sync is
+frought with conflicts that can cause missing changes or just make for a very
+tedious merge.
+
+The idea behind the autosyncing strategy is that the downstream branches will
+match the upstream branches 1 for 1. So downstream `master` will track upstream
+`master`. Downstream `vx.y.z` will track upstream `vx.y.z`. So the syncs should
+occur with minimal conflicts. These downstream branches will NOT contain any
+special downstream fixes, they are strictly mirrors of the upstream.
+
+The autosyncing will occur on a frequent basis configured in the
+[configuration file][auto-config]. The frequency will be determined by our
+needs.
+
+We will manage our own OpenShift release branches. These branches will be a
+combination of a tagged upstream branch plus the overlay branch.
+
+There is an initial prototype of the syncing code already:
+
+https://github.com/fabianvf/ocp-osdk-downstream-automation
+
+#### Overlay Branch Directory Structures
+
+Today, we have the following files in the `ocp-release-operator-sdk` repo. This
+usually doesn't impact the merge but it is a set of files that are NOT upstream.
+So we will put these in an overlay branch that can be used to lay down over the
+mirrored upstream branch to create a downstream release branch.
+
+The following is a list of directories and files that will exist in the overlay
+branch:
+
+* ci
+  * dockerfiles
+    * variety of Dockerfiles used to build images for running tests
+* prow.Makefile # Makefile used by CI
+* patches
+  * contain the patch files that may be carried
+* release
+  * ansible
+    * ansible_collections # contains the sync ansible collections
+    * Dockerfiles # various dockerfiles
+  * helm
+    * Dockerfiles # various dockerfiles
+  * sdk
+    * Dockerfiles # various dockerfiles
+  * scorecard-test
+    * Dockerfiles # various dockerfiles
+  * scorecard-test-kuttl
+    * Dockerfiles # various dockerfiles
+* vendor
+  * contains the dependencies for the project
+
+#### Handling Patches
+
+Typically we want to submit all bugs upstream, have them release then sync the
+release downstream. But there will be times we need to fix a problem downstream
+to resolve a bug that we can’t wait for an upstream release. The process here
+will be twofold.
+
+1. For fixes to vendored dependencies, have a patch in the patches directory
+   that will be applied to the vendor after we run go mod vendor.
+1. For fixes to mainline code, fix the mainline code directly. NOTE: these changes
+   will be lost when re-running UPSTREAM-MERGE.sh script.
```

Comment:
When we merge from upstream and overwrite any downstream-only changes, how do we ensure that there are no regressions?

What if we need to keep the downstream-only changes because a fix was not applied upstream?

Reply:
Rewrote this section because I stole it from the old plan. 
```
Typically we want to submit all bugs upstream, have them release then sync the 
release downstream. But there will be times we need to fix a problem downstream
to resolve a bug that we can’t wait for an upstream release. The process here
will be to keep patches separate to avoid conflicts during syncing.

For fixes to vendored dependencies, have a patch in the patches directory
that will be applied to the vendor after we run go mod vendor.

For fixes to mainline code, have a patch in the patches directory that will be
applied before builds. In order to do this, we may have to create a downstream
`Makefile` that could include the upstream `Makefile` if we need to insert
patches step.

For either of the above, we will always submit the patches to the appropriate
upstream repos.
```